testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/core/JdkModuleDependencyTestCase.java

```diff
@@ -0,0 +1,214 @@
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.deployment.trivial.ServiceActivatorDeploymentUtil;
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.as.test.shared.PermissionUtils;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+import javax.inject.Inject;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.PropertyPermission;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Test case checking whether a module can depend on standard OpenJDK 9+ platform modules (or their emulation in case of Java 8).
+ *
+ * @author <a href="mailto:mjurc@redhat.com">Michal Jurc</a> (c) 2018 Red Hat, Inc.
+ */
+@RunWith(WildflyTestRunner.class)
+public class JdkModuleDependencyTestCase {
+
+    public static final String DEPLOYMENT_NAME_SUFFIX = "-test-dep.jar";
+
+    /**
+     * A set of modules that we need to emulate to be compatible with JDK8.
+     */
+    public static final Set<String> REQUIRED_1_8_EMULATED_MODULES = Stream.of("java.base", "java.compiler", "java.datatransfer",
+            "java.desktop", "java.instrument", "java.jnlp", "java.logging", "java.management", "java.management.rmi",
+            "java.naming", "java.prefs", "java.rmi", "java.scripting", "java.security.jgss", "java.security.sasl",
+            "java.smartcardio", "java.sql", "java.sql.rowset", "java.xml", "java.xml.crypto", "javafx.base", "javafx.controls",
+            "javafx.fxml", "javafx.graphics", "javafx.media", "javafx.swing", "javafx.web", "jdk.accessibility", "jdk.attach",
+            "jdk.compiler", "jdk.httpserver", "jdk.jartool", "jdk.javadoc", "jdk.jconsole", "jdk.jdi", "jdk.jfr", "jdk.jsobject",
+            "jdk.management", "jdk.management.cmm", "jdk.management.jfr", "jdk.management.resource", "jdk.net", "jdk.plugin.dom",
+            "jdk.scripting.nashorn", "jdk.sctp", "jdk.security.auth", "jdk.security.jgss", "jdk.unsupported", "jdk.xml.dom",
+            "org.jboss.modules").collect(Collectors.toCollection(HashSet::new));
+
+    /**
+     * A set of all non-internal JDK9 API modules.
+     *
+     * @see <a href="https://docs.oracle.com/javase/9/docs/api/overview-summary.html">Overview (Java SE 9 &amp; JDK 9 )</a>
```

Comment:
@dmlloyd Would you kindly review this set of modules? I have taken them from official JavaDoc, however, with OpenJDK 9.0.4 I am getting missing following:
```
[ERROR] Failures: 
[ERROR]   JdkModuleDependencyTestCase.testJdk9ModuleDependencies:117->testModuleDependencies:201 Deployments with the following module dependencies have failed: [jdk.management.jfr, javafx.web, java.transaction, jdk.hotspot.agent, java.xml.ws, jdk.pack, java.xml.ws.annotation, javafx.swing, jdk.incubator.httpclient, jdk.jfr, jdk.rmic, jdk.packager.services, java.se.ee, java.xml.bind, javafx.base, jdk.jcmd, java.jnlp, javafx.graphics, javafx.controls, jdk.management.resource, javafx.fxml, javafx.media, jdk.snmp, jdk.management.cmm]
```
The same failure list applies for OpenJDK 10.0.1.

Reply:
The `jdk.*` modules are not guaranteed to exist on any given JDK.  In addition, IIRC Oracle provides some modules that are not found in OpenJDK.

Is it possible to narrow down the failing module?

Reply:
I know that `java.jnlp` is being removed at some point in the future as well.